airflow-core/src/airflow/assets/manager.py

```diff
@@ -54,6 +55,63 @@
 log = structlog.get_logger(__name__)
 
 
+@contextmanager
+def _acquire_apdr_lock(
```

Comment:
you are not actually locking ADPR.  i think it's better to tell the truth.  e.g. _lock_asset(...)